lib/Twig/Extension/Core.php

```diff
@@ -298,17 +298,28 @@ public function parseNotTestExpression(Twig_Parser $parser, Twig_NodeInterface $
     public function parseTestExpression(Twig_Parser $parser, Twig_NodeInterface $node)
     {
         $stream = $parser->getStream();
-        $name = $this->getTestName($parser, $node->getLine());
-        $class = $this->getTestNodeClass($parser, $name);
+        $test = $this->getTest($parser, $node->getLine());
+
+        if ($test->isDeprecated()) {
+            $message = sprintf('Twig Test "%s" is deprecated', $test->getName());
+            if ($test->getAlternative()) {
+                $message .= sprintf('. Use "%s" instead', $test->getAlternative());
+            }
+            $message .= sprintf(' in %s at line %d.', $stream->getFilename(), $stream->getCurrent()->getLine());
+
+            @trigger_error($message, E_USER_DEPRECATED);
+        }
+
+        $class = $this->getTestNodeClass($parser, $test);
         $arguments = null;
         if ($stream->test(Twig_Token::PUNCTUATION_TYPE, '(')) {
             $arguments = $parser->getExpressionParser()->parseArguments(true);
         }
 
-        return new $class($node, $name, $arguments, $parser->getCurrentToken()->getLine());
+        return new $class($node, $test->getName(), $arguments, $parser->getCurrentToken()->getLine());
     }
 
-    protected function getTestName(Twig_Parser $parser, $line)
+    protected function getTest(Twig_Parser $parser, $line)
```

Comment:
I know, but nobody should rely on this and anyway, Twig is not Symfony. That's almost fixes in 2.0 as most methods/properties are now private.